datasets/iq/aml_list/crawler.py

```diff
@@ -0,0 +1,135 @@
+
+
+def extract_listing_date(decision_number: str) -> Optional[str]:
+    """Extracts the listing year from the decision number."""
+    if decision_number is None:
+        return None
+    match = YEAR_PATTERN.search(decision_number)
+    return match.group(0) if match else None
+
+
+def crawl_row(row: dict, context: Context):
+    entity_name = row.pop("asm_alkyan", None)  # اسم الكيان
+    id = row.pop("t")  # ت
+    decision_number = row.pop("rqm_alqrar")  # رقم القرار
+
+    reason = extract_reason(entity_name)
+    entity_name = clean_entity_name(entity_name)
+    listing_date = extract_listing_date(decision_number)
+
+    if entity_name:
+        entity = context.make("LegalEntity")
+        entity.id = context.make_id(id, entity_name, decision_number)
+        entity.add("topics", "debarment")
+        entity.add("name", entity_name, lang="ara")
+        context.emit(entity)
+        sanction = h.make_sanction(context, entity)
+    else:
+        name = row.pop("asm_alshkhs", row.pop("asma_alashkhas", None))  # اسم الشخص
+        person = context.make("Person")
+        person.id = context.make_id(id, name)
+        person.add("topics", "debarment")
+        person.add("nationality", row.pop("aljnsyt", None), lang="ara")  # الجنسية
+        h.apply_date(person, "birthDate", row.pop("altwld"))  # التولد
+        h.apply_name(
+            person,
+            full=name,
+            matronymic=row.pop("asm_alam"),  # اسم الام
+            lang="ara",
+        )
+        context.emit(person)
+        sanction = h.make_sanction(context, person)
+
+    sanction.add("recordId", decision_number)
+    sanction.add("reason", reason, lang="ara")
+    h.apply_date(sanction, "listingDate", listing_date)
+    context.emit(sanction)
+
+    context.audit_data(row)
+
+
+def process_xlsx(
+    context,
+    url: str,
+    filename: str,
+    expected_sheets: list,
+    title: str,
+    ignore_sheets: list = [],
+):
+    doc = context.fetch_html(url, cache_days=1)
+    link = doc.xpath(f'//article[@id="post-{url.split("=")[-1]}"]//a/@href')
+
+    assert len(link) == 1, link
+    file_url = link[0]
+    assert file_url.endswith(".xlsx"), file_url
+    assert title in file_url, file_url
+
+    path = context.fetch_resource(filename, file_url)
+    context.export_resource(path, XLSX)
+
+    wb = load_workbook(path, read_only=True)
+    for sheet in expected_sheets:
+        for row in h.parse_xlsx_sheet(context, wb[sheet], skiprows=3):
+            crawl_row(row, context)
+
+    assert set(wb.sheetnames) == set(expected_sheets + ignore_sheets)
+
+
+def crawl(context: Context):
+    process_xlsx(
+        context,
+        SOURCE_URLS[0],
+        "international.xlsx",
+        INT_LISTS,
+        "القائمة-الدولية",  # International list
+    )
+    process_xlsx(
+        context,
+        SOURCE_URLS[1],
+        "local.xlsx",
+        LOC_PERSON_LISTS + LOC_ENT_LIST,
+        "القوائم-المحلية",  # Local lists
+        LOC_IGNORE_LIST,
+    )
```

Comment:
can we just hardcode the source URLs here? It's so much easier to know what we're talking about. So much harder to break by swapping them in the list and forgetting to check if order matters.